---
title: "bd"
author: "Tord Ranheim"
date: "2025-04-24"
output:
  html_document:
    keep_md: no
---


Beta diversity analyses


```{r setup, include=FALSE}
knitr::opts_chunk$set(dev = "png",
                      dpi=300,
                      echo=F,
                      cache=T, echo=T)
```



                                         LOAD LIBRARIES
###################################################################################################

```{r Load libraries}
library(tidyverse)
library(rmarkdown)
library(knitr)
library(magrittr)
library(ggplot2)
library(RColorBrewer)
library(ggpubr)
library(ggsci)
library(dplyr)
library(phyloseq)
library(microbiome)
library(vegan)
library(metagMisc)
library(MiscMetabar)
library(ape)
library(phylosmith)
library(microViz)
library(pairwiseAdonis)
library(permute)
```



```{r Clear directory and set seed}
rm(list=ls())
set.seed(123)
```



                                         LOAD FUNCTIONS
###################################################################################################

sourced from: https://stackoverflow.com/questions/74446879/how-can-the-effect-size-of-a-permanova-be-calculated

```{r Effect size function}

#' Calculate (partial) Omega-squared (effect-size calculation) for PERMANOVA and add it to the input object
#'
#' @param adonisOutput An adonis object
#' @param partial Should partial omega-squared be calculated (sample size adjusted). Default TRUE
#' @return Original adonis object with the (partial) Omega-squared values added
#' @import vegan
#' @export
adonis_OmegaSq <- function(adonisOutput, partial = TRUE){
    if(!(is(adonisOutput, "adonis") || is(adonisOutput, "anova.cca")))
        stop("Input should be an adonis object")
    if (is(adonisOutput, "anova.cca")) {
        aov_tab <- adonisOutput
        aov_tab$MeanSqs <- aov_tab$SumOfSqs / aov_tab$Df
        aov_tab$MeanSqs[length(aov_tab$Df)] <- NA
    } else {
        aov_tab <- adonisOutput$aov.tab
    }
    heading <- attr(aov_tab, "heading")
    MS_res <- aov_tab[pmatch("Residual", rownames(aov_tab)), "MeanSqs"]
    SS_tot <- aov_tab[rownames(aov_tab) == "Total", "SumsOfSqs"]
    N <- aov_tab[rownames(aov_tab) == "Total", "Df"] + 1
    if(partial){
        omega <- apply(aov_tab, 1, function(x) (x["Df"]*(x["MeanSqs"]-MS_res))/(x["Df"]*x["MeanSqs"]+(N-x["Df"])*MS_res))
        aov_tab$parOmegaSq <- c(omega[1:(length(omega)-2)], NA, NA)
    } else {
        omega <- apply(aov_tab, 1, function(x) (x["SumsOfSqs"]-x["Df"]*MS_res)/(SS_tot+MS_res))
        aov_tab$OmegaSq <- c(omega[1:(length(omega)-2)], NA, NA)
    }
    if (is(adonisOutput, "adonis"))
        cn_order <- c("Df", "SumsOfSqs", "MeanSqs", "F.Model", "R2",
                      if (partial) "parOmegaSq" else "OmegaSq", "Pr(>F)")
    else
        cn_order <- c("Df", "SumOfSqs", "F", if (partial) "parOmegaSq" else "OmegaSq",
                      "Pr(>F)")
    aov_tab <- aov_tab[, cn_order]
    attr(aov_tab, "names") <- cn_order
    attr(aov_tab, "heading") <- heading
    if (is(adonisOutput, "adonis"))
        adonisOutput$aov.tab <- aov_tab
    else
        adonisOutput <- aov_tab
    return(adonisOutput)
}

```


                                         LOAD DATA
###################################################################################################


Data loading
```{r Load phyloseq objects}

#Soil fungi
ps_fungi = readRDS("/Users/tranheim/Desktop/Studier/Remiil-NFI/Study 1/Processing/Microbial/Amplicon/Soil/ITS/Processing/Results/ps_fungi_srs.RDS")

#Soil bacteria
ps_bacteria = readRDS("/Users/tranheim/Desktop/Studier/Remiil-NFI/Study 1/Processing/Microbial/Amplicon/Soil/16S/Processing/Results/ps_bacteria_srs.RDS")


#CAZymes
ps_caz_bac = readRDS("/Users/tranheim/Desktop/Studier/Remiil-NFI/Study 1/Processing/Microbial/Metagenome/Processing/CAZymes/Results/ps_caz_bac_rar.RDS")
ps_caz_fun = readRDS("/Users/tranheim/Desktop/Studier/Remiil-NFI/Study 1/Processing/Microbial/Metagenome/Processing/CAZymes/Results/ps_caz_fun_rar.RDS")


#P-cyc
ps_pcyc_bac = readRDS("/Users/tranheim/Desktop/Studier/Remiil-NFI/Study 1/Processing/Microbial/Metagenome/Processing/Pcyc/Results/ps_pcyc_bac_rar.RDS")
ps_pcyc_fun = readRDS("/Users/tranheim/Desktop/Studier/Remiil-NFI/Study 1/Processing/Microbial/Metagenome/Processing/Pcyc/Results/ps_pcyc_fun_rar.RDS")

#N-cyc
ps_ncyc_bac = readRDS("/Users/tranheim/Desktop/Studier/Remiil-NFI/Study 1/Processing/Microbial/Metagenome/Processing/Ncyc/Results/ps_ncyc_bac_rar.RDS")



```


```{r Load metadata file}

metadata = read.table("/Users/tranheim/Desktop/Studier/Remiil-NFI/Study 1/Processing/Metadata/Result/metadata.txt")
row.names(metadata) = metadata$Sample

```



                                         PERMANOVA TESTS
###########################################################################################

Now set up new tests with permutations restricted to pairs to account for matched study design

```{r Pairwise comparisons fungi without covariates}

#Pairwise comparisons of land-use stages using permanova
#As grassland sites are not matched, these will be evaluated separately 

ps_fungi_grass = ps_fungi %>% ps_filter(Succession != "Forest") #Subset ps to grassland sites only

groups <- as.data.frame(t(combn(unique(as.character(ps_fungi_grass@sam_data$Succession)), m = 2))) #Set contrasts

contrasts_g <- data.frame(
    group1 = groups$V1, group2 = groups$V2,
    R2 = NA, F_value = NA, df1 = NA, df2 = NA, p_value = NA
  ) #Create dataframe to add results to


#Loop through pairwise grassland comparisons and add results to data frame
for (i in seq(nrow(contrasts_g))) {
ps <- ps_fungi_grass %>% 
ps_filter(Succession %in% c(contrasts_g[i,1],contrasts_g[i,2]))#Subset phyloseq to contrasts

dm = ps %>% phyloseq::distance(method = "bray") #Calculate distance matrix

fit <- vegan::adonis2(dm ~ sample_data(ps)$Succession, 
                      nperm=9999,
                      method = "bray")

   contrasts_g$R2[i] <- round(fit$R2[1], digits = 3)
    contrasts_g$F_value[i] <- round(fit[["F"]][1], digits = 3)
    contrasts_g$df1[i] <- fit$Df[1]
    contrasts_g$df2[i] <- fit$Df[2]
    contrasts_g$p_value[i] <- fit$`Pr(>F)`[1]
}


#For pairwise comparisons between grassland and forest sites we create a separate loop that accounts for the paired structure of the dataset
groups <- as.data.frame(t(combn(unique(as.character(sample_data(ps_fungi)$Succession)), m = 2)))
groups <- groups[1:3,] #Restrict contrasts to grassland-forest comparisons

contrasts <- data.frame(
    group1 = groups$V1, group2 = groups$V2,
    R2 = NA, F_value = NA, df1 = NA, df2 = NA, p_value = NA
  )


#Loop over dataset to make pairwise permanova tests
for (i in seq(nrow(contrasts))) {

ps <- ps_fungi %>% 
ps_filter(Succession %in% c(contrasts[i,1],contrasts[i,2])) %>% #Subset phyloseq to contrasts
ps_filter(n()>1, .by = Par)  #Filter out unmatched observations

dm = ps %>% phyloseq::distance(method = "bray") #Calculate distance matrix

h <- with(sample_data(ps),how(blocks = Par, nperm = 9999)) #Restrict permutations

fit <- vegan::adonis2(dm ~ sample_data(ps)$Succession, 
                      permutation = h,
                      method = "bray")

   contrasts$R2[i] <- round(fit$R2[1], digits = 3)
    contrasts$F_value[i] <- round(fit[["F"]][1], digits = 3)
    contrasts$df1[i] <- fit$Df[1]
    contrasts$df2[i] <- fit$Df[2]
    contrasts$p_value[i] <- fit$`Pr(>F)`[1]
}


#Merge dataframes
s1 = rbind(contrasts,contrasts_g)
s1$p_value_adjusted = p.adjust(s1$p_value, method = "BH") #Adjust p-values for multiple comparisons
s1$Covariates = 0
s1$Community = "Fungi"
s1$Diversity = "Taxonomic"
r1 = s1

```



```{r Pairwise comparisons bacteria without covariates}

#Pairwise comparisons of land-use stages using permanova
#As grassland sites are not matched, these will be evaluated separately 

ps_bac_grass = ps_bacteria %>% ps_filter(Succession != "Forest") #Subset ps to grassland sites only

groups <- as.data.frame(t(combn(unique(as.character(ps_bac_grass@sam_data$Succession)), m = 2))) #Set contrasts

contrasts_g <- data.frame(
    group1 = groups$V1, group2 = groups$V2,
    R2 = NA, F_value = NA, df1 = NA, df2 = NA, p_value = NA
  ) #Create dataframe to add results to


#Loop through pairwise grassland comparisons and add results to data frame
for (i in seq(nrow(contrasts_g))) {
ps <- ps_bac_grass %>% 
ps_filter(Succession %in% c(contrasts_g[i,1],contrasts_g[i,2]))#Subset phyloseq to contrasts

dm = ps %>% phyloseq::distance(method = "bray") #Calculate distance matrix


fit <- vegan::adonis2(dm ~ sample_data(ps)$Succession, 
                      nperm=9999,
                      method = "bray")

   contrasts_g$R2[i] <- round(fit$R2[1], digits = 3)
    contrasts_g$F_value[i] <- round(fit[["F"]][1], digits = 3)
    contrasts_g$df1[i] <- fit$Df[1]
    contrasts_g$df2[i] <- fit$Df[2]
    contrasts_g$p_value[i] <- fit$`Pr(>F)`[1]
}


#For pairwise comparisons between grassland and forest sites we create a separate loop that accounts for the paired structure of the dataset
groups <- as.data.frame(t(combn(unique(as.character(sample_data(ps_bacteria)$Succession)), m = 2)))
groups <- groups[1:3,] #Restrict contrasts to grassland-forest comparisons

contrasts <- data.frame(
    group1 = groups$V1, group2 = groups$V2,
    R2 = NA, F_value = NA, df1 = NA, df2 = NA, p_value = NA
  )


#Loop over dataset to make pairwise permanova tests
for (i in seq(nrow(contrasts))) {

ps <- ps_bacteria %>% 
ps_filter(Succession %in% c(contrasts[i,1],contrasts[i,2])) %>% #Subset phyloseq to contrasts
ps_filter(n()>1, .by = Par)  #Filter out unmatched observations

dm = ps %>% phyloseq::distance(method = "bray") #Calculate distance matrix

h <- with(sample_data(ps),how(blocks = Par, nperm = 9999)) #Restrict permutations

fit <- vegan::adonis2(dm ~ sample_data(ps)$Succession, 
                      permutation = h,
                      method = "bray")

   contrasts$R2[i] <- round(fit$R2[1], digits = 3)
    contrasts$F_value[i] <- round(fit[["F"]][1], digits = 3)
    contrasts$df1[i] <- fit$Df[1]
    contrasts$df2[i] <- fit$Df[2]
    contrasts$p_value[i] <- fit$`Pr(>F)`[1]
}


#Merge dataframes
s1 = rbind(contrasts,contrasts_g)
s1$p_value_adjusted = p.adjust(s1$p_value, method = "BH") #Adjust p-values for multiple comparisons
s1$Covariates = 0
s1$Community = "Bacteria"
s1$Diversity = "Taxonomic"
r2 = s1

```


```{r Fungi cazymes}

#Pairwise comparisons of land-use stages using permanova
#As grassland sites are not matched, these will be evaluated separately 

p1 = ps_caz_fun %>% ps_filter(Succession != "Forest") #Subset ps to grassland sites only

groups <- as.data.frame(t(combn(unique(as.character(p1@sam_data$Succession)), m = 2))) #Set contrasts

contrasts_g <- data.frame(
    group1 = groups$V1, group2 = groups$V2,
    R2 = NA, F_value = NA, df1 = NA, df2 = NA, p_value = NA
  ) 


#Loop through pairwise grassland comparisons and add results to data frame
for (i in seq(nrow(contrasts_g))) {
ps <- p1 %>% 
ps_filter(Succession %in% c(contrasts_g[i,1],contrasts_g[i,2]))#Subset phyloseq to contrasts

dm = ps %>% phyloseq::distance(method = "bray") #Calculate distance matrix

fit <- vegan::adonis2(dm ~ sample_data(ps)$Succession, 
                      nperm=9999,
                      method = "bray")

   contrasts_g$R2[i] <- round(fit$R2[1], digits = 3)
    contrasts_g$F_value[i] <- round(fit[["F"]][1], digits = 3)
    contrasts_g$df1[i] <- fit$Df[1]
    contrasts_g$df2[i] <- fit$Df[2]
    contrasts_g$p_value[i] <- fit$`Pr(>F)`[1]
}


#For pairwise comparisons between grassland and forest sites we create a separate loop that accounts for the paired structure of the dataset
groups <- as.data.frame(t(combn(unique(as.character(sample_data(ps_fungi)$Succession)), m = 2)))
groups <- groups[1:3,] #Restrict contrasts to grassland-forest comparisons

contrasts <- data.frame(
    group1 = groups$V1, group2 = groups$V2,
    R2 = NA, F_value = NA, df1 = NA, df2 = NA, p_value = NA
  )


#Loop over dataset to make pairwise permanova tests
for (i in seq(nrow(contrasts))) {

ps <- ps_caz_fun %>% 
ps_filter(Succession %in% c(contrasts[i,1],contrasts[i,2])) %>% #Subset phyloseq to contrasts
ps_filter(n()>1, .by = Par)  #Filter out unmatched observations

dm = ps %>% phyloseq::distance(method = "bray") #Calculate distance matrix

h <- with(sample_data(ps),how(blocks = Par, nperm = 9999)) #Restrict permutations

fit <- vegan::adonis2(dm ~ sample_data(ps)$Succession, 
                      permutation = h,
                      method = "bray")

   contrasts$R2[i] <- round(fit$R2[1], digits = 3)
    contrasts$F_value[i] <- round(fit[["F"]][1], digits = 3)
    contrasts$df1[i] <- fit$Df[1]
    contrasts$df2[i] <- fit$Df[2]
    contrasts$p_value[i] <- fit$`Pr(>F)`[1]
}


#Merge dataframes
s1 = rbind(contrasts,contrasts_g)
s1$p_value_adjusted = p.adjust(s1$p_value, method = "BH") #Adjust p-values for multiple comparisons
s1$Covariates = 0
s1$Community = "Fungi"
s1$Diversity = "C-cycling"
r3 = s1

```


```{r Fungi pcyc}

#Pairwise comparisons of land-use stages using permanova
#As grassland sites are not matched, these will be evaluated separately 

p1 = ps_pcyc_fun %>% ps_filter(Succession != "Forest") #Subset ps to grassland sites only

groups <- as.data.frame(t(combn(unique(as.character(p1@sam_data$Succession)), m = 2))) #Set contrasts

contrasts_g <- data.frame(
    group1 = groups$V1, group2 = groups$V2,
    R2 = NA, F_value = NA, df1 = NA, df2 = NA, p_value = NA
  ) 


#Loop through pairwise grassland comparisons and add results to data frame
for (i in seq(nrow(contrasts_g))) {
ps <- p1 %>% 
ps_filter(Succession %in% c(contrasts_g[i,1],contrasts_g[i,2]))#Subset phyloseq to contrasts

dm = ps %>% phyloseq::distance(method = "bray") #Calculate distance matrix

fit <- vegan::adonis2(dm ~ sample_data(ps)$Succession, 
                      nperm=9999,
                      method = "bray")

   contrasts_g$R2[i] <- round(fit$R2[1], digits = 3)
    contrasts_g$F_value[i] <- round(fit[["F"]][1], digits = 3)
    contrasts_g$df1[i] <- fit$Df[1]
    contrasts_g$df2[i] <- fit$Df[2]
    contrasts_g$p_value[i] <- fit$`Pr(>F)`[1]
}


#For pairwise comparisons between grassland and forest sites we create a separate loop that accounts for the paired structure of the dataset
groups <- as.data.frame(t(combn(unique(as.character(sample_data(ps_fungi)$Succession)), m = 2)))
groups <- groups[1:3,] #Restrict contrasts to grassland-forest comparisons

contrasts <- data.frame(
    group1 = groups$V1, group2 = groups$V2,
    R2 = NA, F_value = NA, df1 = NA, df2 = NA, p_value = NA
  )


#Loop over dataset to make pairwise permanova tests
for (i in seq(nrow(contrasts))) {

ps <- ps_pcyc_fun %>% 
ps_filter(Succession %in% c(contrasts[i,1],contrasts[i,2])) %>% #Subset phyloseq to contrasts
ps_filter(n()>1, .by = Par)  #Filter out unmatched observations

dm = ps %>% phyloseq::distance(method = "bray") #Calculate distance matrix

h <- with(sample_data(ps),how(blocks = Par, nperm = 9999)) #Restrict permutations

fit <- vegan::adonis2(dm ~ sample_data(ps)$Succession, 
                      permutation = h,
                      method = "bray")

   contrasts$R2[i] <- round(fit$R2[1], digits = 3)
    contrasts$F_value[i] <- round(fit[["F"]][1], digits = 3)
    contrasts$df1[i] <- fit$Df[1]
    contrasts$df2[i] <- fit$Df[2]
    contrasts$p_value[i] <- fit$`Pr(>F)`[1]
}


#Merge dataframes
s1 = rbind(contrasts,contrasts_g)
s1$p_value_adjusted = p.adjust(s1$p_value, method = "BH") #Adjust p-values for multiple comparisons
s1$Covariates = 0
s1$Community = "Fungi"
s1$Diversity = "P-cycling"
r4 = s1

```



```{r Bacteria cazymes}

#Pairwise comparisons of land-use stages using permanova
#As grassland sites are not matched, these will be evaluated separately 

p1 = ps_caz_bac %>% ps_filter(Succession != "Forest") #Subset ps to grassland sites only

groups <- as.data.frame(t(combn(unique(as.character(p1@sam_data$Succession)), m = 2))) #Set contrasts

contrasts_g <- data.frame(
    group1 = groups$V1, group2 = groups$V2,
    R2 = NA, F_value = NA, df1 = NA, df2 = NA, p_value = NA
  ) 


#Loop through pairwise grassland comparisons and add results to data frame
for (i in seq(nrow(contrasts_g))) {
ps <- p1 %>% 
ps_filter(Succession %in% c(contrasts_g[i,1],contrasts_g[i,2]))#Subset phyloseq to contrasts

dm = ps %>% phyloseq::distance(method = "bray") #Calculate distance matrix

fit <- vegan::adonis2(dm ~ sample_data(ps)$Succession, 
                      nperm=9999,
                      method = "bray")

   contrasts_g$R2[i] <- round(fit$R2[1], digits = 3)
    contrasts_g$F_value[i] <- round(fit[["F"]][1], digits = 3)
    contrasts_g$df1[i] <- fit$Df[1]
    contrasts_g$df2[i] <- fit$Df[2]
    contrasts_g$p_value[i] <- fit$`Pr(>F)`[1]
}


#For pairwise comparisons between grassland and forest sites we create a separate loop that accounts for the paired structure of the dataset
groups <- as.data.frame(t(combn(unique(as.character(sample_data(ps_fungi)$Succession)), m = 2)))
groups <- groups[1:3,] #Restrict contrasts to grassland-forest comparisons

contrasts <- data.frame(
    group1 = groups$V1, group2 = groups$V2,
    R2 = NA, F_value = NA, df1 = NA, df2 = NA, p_value = NA
  )


#Loop over dataset to make pairwise permanova tests
for (i in seq(nrow(contrasts))) {

ps <- ps_caz_bac %>% 
ps_filter(Succession %in% c(contrasts[i,1],contrasts[i,2])) %>% #Subset phyloseq to contrasts
ps_filter(n()>1, .by = Par)  #Filter out unmatched observations

dm = ps %>% phyloseq::distance(method = "bray") #Calculate distance matrix

h <- with(sample_data(ps),how(blocks = Par, nperm = 9999)) #Restrict permutations

fit <- vegan::adonis2(dm ~ sample_data(ps)$Succession, 
                      permutation = h,
                      method = "bray")

   contrasts$R2[i] <- round(fit$R2[1], digits = 3)
    contrasts$F_value[i] <- round(fit[["F"]][1], digits = 3)
    contrasts$df1[i] <- fit$Df[1]
    contrasts$df2[i] <- fit$Df[2]
    contrasts$p_value[i] <- fit$`Pr(>F)`[1]
}


#Merge dataframes
s1 = rbind(contrasts,contrasts_g)
s1$p_value_adjusted = p.adjust(s1$p_value, method = "BH") #Adjust p-values for multiple comparisons
s1$Covariates = 0
s1$Community = "Bacteria"
s1$Diversity = "C-cycling"
r5 = s1

```


```{r Bacteria pcyc}

#Pairwise comparisons of land-use stages using permanova
#As grassland sites are not matched, these will be evaluated separately 

p1 = ps_pcyc_bac %>% ps_filter(Succession != "Forest") #Subset ps to grassland sites only

groups <- as.data.frame(t(combn(unique(as.character(p1@sam_data$Succession)), m = 2))) #Set contrasts

contrasts_g <- data.frame(
    group1 = groups$V1, group2 = groups$V2,
    R2 = NA, F_value = NA, df1 = NA, df2 = NA, p_value = NA
  ) 


#Loop through pairwise grassland comparisons and add results to data frame
for (i in seq(nrow(contrasts_g))) {
ps <- p1 %>% 
ps_filter(Succession %in% c(contrasts_g[i,1],contrasts_g[i,2]))#Subset phyloseq to contrasts

dm = ps %>% phyloseq::distance(method = "bray") #Calculate distance matrix

fit <- vegan::adonis2(dm ~ sample_data(ps)$Succession, 
                      nperm=9999,
                      method = "bray")

   contrasts_g$R2[i] <- round(fit$R2[1], digits = 3)
    contrasts_g$F_value[i] <- round(fit[["F"]][1], digits = 3)
    contrasts_g$df1[i] <- fit$Df[1]
    contrasts_g$df2[i] <- fit$Df[2]
    contrasts_g$p_value[i] <- fit$`Pr(>F)`[1]
}


#For pairwise comparisons between grassland and forest sites we create a separate loop that accounts for the paired structure of the dataset
groups <- as.data.frame(t(combn(unique(as.character(sample_data(ps_fungi)$Succession)), m = 2)))
groups <- groups[1:3,] #Restrict contrasts to grassland-forest comparisons

contrasts <- data.frame(
    group1 = groups$V1, group2 = groups$V2,
    R2 = NA, F_value = NA, df1 = NA, df2 = NA, p_value = NA
  )


#Loop over dataset to make pairwise permanova tests
for (i in seq(nrow(contrasts))) {

ps <- ps_pcyc_bac %>% 
ps_filter(Succession %in% c(contrasts[i,1],contrasts[i,2])) %>% #Subset phyloseq to contrasts
ps_filter(n()>1, .by = Par)  #Filter out unmatched observations

dm = ps %>% phyloseq::distance(method = "bray") #Calculate distance matrix

h <- with(sample_data(ps),how(blocks = Par, nperm = 9999)) #Restrict permutations

fit <- vegan::adonis2(dm ~ sample_data(ps)$Succession, 
                      permutation = h,
                      method = "bray")

   contrasts$R2[i] <- round(fit$R2[1], digits = 3)
    contrasts$F_value[i] <- round(fit[["F"]][1], digits = 3)
    contrasts$df1[i] <- fit$Df[1]
    contrasts$df2[i] <- fit$Df[2]
    contrasts$p_value[i] <- fit$`Pr(>F)`[1]
}


#Merge dataframes
s1 = rbind(contrasts,contrasts_g)
s1$p_value_adjusted = p.adjust(s1$p_value, method = "BH") #Adjust p-values for multiple comparisons
s1$Covariates = 0
s1$Community = "Bacteria"
s1$Diversity = "P-cycling"
r6 = s1

```



```{r Bacteria ncyc}

#Pairwise comparisons of land-use stages using permanova
#As grassland sites are not matched, these will be evaluated separately 

p1 = ps_ncyc_bac %>% ps_filter(Succession != "Forest") #Subset ps to grassland sites only

groups <- as.data.frame(t(combn(unique(as.character(p1@sam_data$Succession)), m = 2))) #Set contrasts

contrasts_g <- data.frame(
    group1 = groups$V1, group2 = groups$V2,
    R2 = NA, F_value = NA, df1 = NA, df2 = NA, p_value = NA
  ) 


#Loop through pairwise grassland comparisons and add results to data frame
for (i in seq(nrow(contrasts_g))) {
ps <- p1 %>% 
ps_filter(Succession %in% c(contrasts_g[i,1],contrasts_g[i,2]))#Subset phyloseq to contrasts

dm = ps %>% phyloseq::distance(method = "bray") #Calculate distance matrix

fit <- vegan::adonis2(dm ~ sample_data(ps)$Succession, 
                      nperm=9999,
                      method = "bray")

   contrasts_g$R2[i] <- round(fit$R2[1], digits = 3)
    contrasts_g$F_value[i] <- round(fit[["F"]][1], digits = 3)
    contrasts_g$df1[i] <- fit$Df[1]
    contrasts_g$df2[i] <- fit$Df[2]
    contrasts_g$p_value[i] <- fit$`Pr(>F)`[1]
}


#For pairwise comparisons between grassland and forest sites we create a separate loop that accounts for the paired structure of the dataset
groups <- as.data.frame(t(combn(unique(as.character(sample_data(ps_fungi)$Succession)), m = 2)))
groups <- groups[1:3,] #Restrict contrasts to grassland-forest comparisons

contrasts <- data.frame(
    group1 = groups$V1, group2 = groups$V2,
    R2 = NA, F_value = NA, df1 = NA, df2 = NA, p_value = NA
  )


#Loop over dataset to make pairwise permanova tests
for (i in seq(nrow(contrasts))) {

ps <- ps_ncyc_bac %>% 
ps_filter(Succession %in% c(contrasts[i,1],contrasts[i,2])) %>% #Subset phyloseq to contrasts
ps_filter(n()>1, .by = Par)  #Filter out unmatched observations

dm = ps %>% phyloseq::distance(method = "bray") #Calculate distance matrix

h <- with(sample_data(ps),how(blocks = Par, nperm = 9999)) #Restrict permutations

fit <- vegan::adonis2(dm ~ sample_data(ps)$Succession, 
                      permutation = h,
                      method = "bray")

   contrasts$R2[i] <- round(fit$R2[1], digits = 3)
    contrasts$F_value[i] <- round(fit[["F"]][1], digits = 3)
    contrasts$df1[i] <- fit$Df[1]
    contrasts$df2[i] <- fit$Df[2]
    contrasts$p_value[i] <- fit$`Pr(>F)`[1]
}


#Merge dataframes
s1 = rbind(contrasts,contrasts_g)
s1$p_value_adjusted = p.adjust(s1$p_value, method = "BH") #Adjust p-values for multiple comparisons
s1$Covariates = 0
s1$Community = "Bacteria"
s1$Diversity = "N-cycling"
r7 = s1

```


```{r}

res_permanova = rbind(r1,r2,r3,r4,r5,r6,r7)

write.table(res_permanova, file = "./results/res_permanova.txt")

rm(r1,r2,r3,r4,r5,r6,r7)
```



                                        CALCULATE EFFECT SIZE
###################################################################################################

```{r}

factors = c("Succession","Residual","Total")
factors2 = c("Succession","pH","C:N","Parent material", "LDMC","Residual","Total")

```


```{r effect size fungi taxonomic}

#WITHOUT COVARIATES
f1 = ps_fungi %>% 
  ps_filter(Succession %in% c("Managed","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")

s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
s1
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Fungi"
q1$R2 = s1$R2
q1$Comparison = "Managed vs Forest"
q1$Diversity = "Taxonomic"
q1$Factor = factors
q1$Covariates = 0
k1 = q1

#Recent
f1 = ps_fungi %>% 
  ps_filter(Succession %in% c("Recent","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Fungi"
q1$R2 = s1$R2
q1$Comparison = "Recent vs Forest"
q1$Diversity = "Taxonomic"
q1$Factor = factors
q1$Covariates = 0
k2 = q1

#Late
f1 = ps_fungi %>% 
  ps_filter(Succession %in% c("Late","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Fungi"
q1$R2 = s1$R2
q1$Comparison = "Late vs Forest"
q1$Diversity = "Taxonomic"
q1$Factor = factors
q1$Covariates = 0
k3 = q1


#WITH COVARIATES
#Managed
f1 = ps_fungi %>% 
  ps_filter(Succession %in% c("Managed","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")

s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession +
                      sample_data(f1)$pH+ 
                      sample_data(f1)$C.N +
                      sample_data(f1)$Parent_material+
                      sample_data(f1)$LDMC,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
s1
q1 = adonis_OmegaSq(s1, partial = T)
q1$R2 = s1$R2
q1$Community = "Fungi"
q1$Comparison = "Managed vs Forest"
q1$Diversity = "Taxonomic"
q1$Factor = factors2
q1$Covariates = 1
k4 = q1

#Recent
f1 = ps_fungi %>% 
  ps_filter(Succession %in% c("Recent","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession +
                      sample_data(f1)$pH+ 
                      sample_data(f1)$C.N +
                      sample_data(f1)$Parent_material+
                      sample_data(f1)$LDMC,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$R2 = s1$R2
q1$Community = "Fungi"
q1$Comparison = "Recent vs Forest"
q1$Diversity = "Taxonomic"
q1$Factor = factors2
q1$Covariates = 1
k5 = q1

#Late
f1 = ps_fungi %>% 
  ps_filter(Succession %in% c("Late","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession +
                      sample_data(f1)$pH+ 
                      sample_data(f1)$C.N +
                      sample_data(f1)$Parent_material+
                      sample_data(f1)$LDMC,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Fungi"
q1$R2 = s1$R2
q1$Comparison = "Late vs Forest"
q1$Diversity = "Taxonomic"
q1$Factor = factors2
q1$Covariates = 1
k6 = q1


r1 = rbind(k1,k2,k3,k4,k5,k6)

```



```{r effect size fungi ccycling}

#WITHOUT COVARIATES
f1 = ps_caz_fun %>% 
  ps_filter(Succession %in% c("Managed","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")

s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
s1
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Fungi"
q1$R2 = s1$R2
q1$Comparison = "Managed vs Forest"
q1$Diversity = "C-cycling"
q1$Factor = factors
q1$Covariates = 0
k1 = q1

#Recent
f1 = ps_caz_fun %>% 
  ps_filter(Succession %in% c("Recent","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Fungi"
q1$R2 = s1$R2
q1$Comparison = "Recent vs Forest"
q1$Diversity = "C-cycling"
q1$Factor = factors
q1$Covariates = 0
k2 = q1

#Late
f1 = ps_caz_fun %>% 
  ps_filter(Succession %in% c("Late","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Fungi"
q1$R2 = s1$R2
q1$Comparison = "Late vs Forest"
q1$Diversity = "C-cycling"
q1$Factor = factors
q1$Covariates = 0
k3 = q1


#WITH COVARIATES
#Managed
f1 = ps_caz_fun %>% 
  ps_filter(Succession %in% c("Managed","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")

s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession +
                      sample_data(f1)$pH+ 
                      sample_data(f1)$C.N +
                      sample_data(f1)$Parent_material+
                      sample_data(f1)$LDMC,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
s1
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Fungi"
q1$R2 = s1$R2
q1$Comparison = "Managed vs Forest"
q1$Diversity = "C-cycling"
q1$Factor = factors2
q1$Covariates = 1
k4 = q1

#Recent
f1 = ps_caz_fun %>% 
  ps_filter(Succession %in% c("Recent","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession +
                      sample_data(f1)$pH+ 
                      sample_data(f1)$C.N +
                      sample_data(f1)$Parent_material+
                      sample_data(f1)$LDMC,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Fungi"
q1$R2 = s1$R2
q1$Comparison = "Recent vs Forest"
q1$Diversity = "C-cycling"
q1$Factor = factors2
q1$Covariates = 1
k5 = q1

#Late
f1 = ps_caz_fun %>% 
  ps_filter(Succession %in% c("Late","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession +
                      sample_data(f1)$pH+ 
                      sample_data(f1)$C.N +
                      sample_data(f1)$Parent_material+
                      sample_data(f1)$LDMC,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Fungi"
q1$R2 = s1$R2
q1$Comparison = "Late vs Forest"
q1$Diversity = "C-cycling"
q1$Factor = factors2
q1$Covariates = 1
k6 = q1


r2 = rbind(k1,k2,k3,k4,k5,k6)

```




```{r effect size fungi pcycling}

#WITHOUT COVARIATES
f1 = ps_pcyc_fun %>% 
  ps_filter(Succession %in% c("Managed","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")

s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
s1
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Fungi"
q1$R2 = s1$R2
q1$Comparison = "Managed vs Forest"
q1$Diversity = "P-cycling"
q1$Factor = factors
q1$Covariates = 0
k1 = q1

#Recent
f1 = ps_pcyc_fun %>% 
  ps_filter(Succession %in% c("Recent","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Fungi"
q1$R2 = s1$R2
q1$Comparison = "Recent vs Forest"
q1$Diversity = "P-cycling"
q1$Factor = factors
q1$Covariates = 0
k2 = q1

#Late
f1 = ps_pcyc_fun %>% 
  ps_filter(Succession %in% c("Late","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Fungi"
q1$R2 = s1$R2
q1$Comparison = "Late vs Forest"
q1$Diversity = "P-cycling"
q1$Factor = factors
q1$Covariates = 0
k3 = q1


#WITH COVARIATES
#Managed
f1 = ps_pcyc_fun %>% 
  ps_filter(Succession %in% c("Managed","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")

s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession +
                      sample_data(f1)$pH+ 
                      sample_data(f1)$C.N +
                      sample_data(f1)$Parent_material+
                      sample_data(f1)$LDMC,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
s1
q1 = adonis_OmegaSq(s1, partial = T)
q1$R2 = s1$R2
q1$Community = "Fungi"
q1$Comparison = "Managed vs Forest"
q1$Diversity = "P-cycling"
q1$Factor = factors2
q1$Covariates = 1
k4 = q1

#Recent
f1 = ps_pcyc_fun %>% 
  ps_filter(Succession %in% c("Recent","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession +
                      sample_data(f1)$pH+ 
                      sample_data(f1)$C.N +
                      sample_data(f1)$Parent_material+
                      sample_data(f1)$LDMC,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Fungi"
q1$R2 = s1$R2
q1$Comparison = "Recent vs Forest"
q1$Diversity = "P-cycling"
q1$Factor = factors2
q1$Covariates = 1
k5 = q1

#Late
f1 = ps_pcyc_fun %>% 
  ps_filter(Succession %in% c("Late","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession +
                      sample_data(f1)$pH+ 
                      sample_data(f1)$C.N +
                      sample_data(f1)$Parent_material+
                      sample_data(f1)$LDMC,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Fungi"
q1$R2 = s1$R2
q1$Comparison = "Late vs Forest"
q1$Diversity = "P-cycling"
q1$Factor = factors2
q1$Covariates = 1
k6 = q1


r3 = rbind(k1,k2,k3,k4,k5,k6)

```





```{r effect size bacteria taxonomic}

#WITHOUT COVARIATES
f1 = ps_bacteria %>% 
  ps_filter(Succession %in% c("Managed","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")

s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
s1
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Managed vs Forest"
q1$Diversity = "Taxonomic"
q1$Factor = factors
q1$Covariates = 0
k1 = q1

#Recent
f1 = ps_bacteria %>% 
  ps_filter(Succession %in% c("Recent","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Recent vs Forest"
q1$Diversity = "Taxonomic"
q1$Factor = factors
q1$Covariates = 0
k2 = q1

#Late
f1 = ps_bacteria %>% 
  ps_filter(Succession %in% c("Late","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Late vs Forest"
q1$Diversity = "Taxonomic"
q1$Factor = factors
q1$Covariates = 0
k3 = q1


#WITH COVARIATES
#Managed
f1 = ps_bacteria %>% 
  ps_filter(Succession %in% c("Managed","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")

s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession +
                      sample_data(f1)$pH+ 
                      sample_data(f1)$C.N +
                      sample_data(f1)$Parent_material+
                      sample_data(f1)$LDMC,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
s1
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Managed vs Forest"
q1$Diversity = "Taxonomic"
q1$Factor = factors2
q1$Covariates = 1
k4 = q1

#Recent
f1 = ps_bacteria %>% 
  ps_filter(Succession %in% c("Recent","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession +
                      sample_data(f1)$pH+ 
                      sample_data(f1)$C.N +
                      sample_data(f1)$Parent_material+
                      sample_data(f1)$LDMC,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Recent vs Forest"
q1$Diversity = "Taxonomic"
q1$Factor = factors2
q1$Covariates = 1
k5 = q1

#Late
f1 = ps_bacteria %>% 
  ps_filter(Succession %in% c("Late","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession +
                      sample_data(f1)$pH+ 
                      sample_data(f1)$C.N +
                      sample_data(f1)$Parent_material+
                      sample_data(f1)$LDMC,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Late vs Forest"
q1$Diversity = "Taxonomic"
q1$Factor = factors2
q1$Covariates = 1
k6 = q1


r4 = rbind(k1,k2,k3,k4,k5,k6)

```


```{r effect size bacteria ccycling}

#WITHOUT COVARIATES
f1 = ps_caz_bac %>% 
  ps_filter(Succession %in% c("Managed","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")

s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
s1
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Managed vs Forest"
q1$Diversity = "C-cycling"
q1$Factor = factors
q1$Covariates = 0
k1 = q1

#Recent
f1 = ps_caz_bac %>% 
  ps_filter(Succession %in% c("Recent","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Recent vs Forest"
q1$Diversity = "C-cycling"
q1$Factor = factors
q1$Covariates = 0
k2 = q1

#Late
f1 = ps_caz_bac %>% 
  ps_filter(Succession %in% c("Late","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Late vs Forest"
q1$Diversity = "C-cycling"
q1$Factor = factors
q1$Covariates = 0
k3 = q1


#WITH COVARIATES
#Managed
f1 = ps_caz_bac %>% 
  ps_filter(Succession %in% c("Managed","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")

s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession +
                      sample_data(f1)$pH+ 
                      sample_data(f1)$C.N +
                      sample_data(f1)$Parent_material+
                      sample_data(f1)$LDMC,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
s1
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Managed vs Forest"
q1$Diversity = "C-cycling"
q1$Factor = factors2
q1$Covariates = 1
k4 = q1

#Recent
f1 = ps_caz_bac %>% 
  ps_filter(Succession %in% c("Recent","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession +
                      sample_data(f1)$pH+ 
                      sample_data(f1)$C.N +
                      sample_data(f1)$Parent_material+
                      sample_data(f1)$LDMC,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Recent vs Forest"
q1$Diversity = "C-cycling"
q1$Factor = factors2
q1$Covariates = 1
k5 = q1

#Late
f1 = ps_caz_bac %>% 
  ps_filter(Succession %in% c("Late","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession +
                      sample_data(f1)$pH+ 
                      sample_data(f1)$C.N +
                      sample_data(f1)$Parent_material+
                      sample_data(f1)$LDMC,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Late vs Forest"
q1$Diversity = "C-cycling"
q1$Factor = factors2
q1$Covariates = 1
k6 = q1


r5 = rbind(k1,k2,k3,k4,k5,k6)

```




```{r effect size bacteria pcycling}

#WITHOUT COVARIATES
f1 = ps_pcyc_bac %>% 
  ps_filter(Succession %in% c("Managed","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")

s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
s1
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Managed vs Forest"
q1$Diversity = "P-cycling"
q1$Factor = factors
q1$Covariates = 0
k1 = q1

#Recent
f1 = ps_pcyc_bac %>% 
  ps_filter(Succession %in% c("Recent","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Recent vs Forest"
q1$Diversity = "P-cycling"
q1$Factor = factors
q1$Covariates = 0
k2 = q1

#Late
f1 = ps_pcyc_bac %>% 
  ps_filter(Succession %in% c("Late","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Late vs Forest"
q1$Diversity = "P-cycling"
q1$Factor = factors
q1$Covariates = 0
k3 = q1


#WITH COVARIATES
#Managed
f1 = ps_pcyc_bac %>% 
  ps_filter(Succession %in% c("Managed","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")

s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession +
                      sample_data(f1)$pH+ 
                      sample_data(f1)$C.N +
                      sample_data(f1)$Parent_material+
                      sample_data(f1)$LDMC,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
s1
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Managed vs Forest"
q1$Diversity = "P-cycling"
q1$Factor = factors2
q1$Covariates = 1
k4 = q1

#Recent
f1 = ps_pcyc_bac %>% 
  ps_filter(Succession %in% c("Recent","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession +
                      sample_data(f1)$pH+ 
                      sample_data(f1)$C.N +
                      sample_data(f1)$Parent_material+
                      sample_data(f1)$LDMC,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Recent vs Forest"
q1$Diversity = "P-cycling"
q1$Factor = factors2
q1$Covariates = 1
k5 = q1

#Late
f1 = ps_pcyc_bac %>% 
  ps_filter(Succession %in% c("Late","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession +
                      sample_data(f1)$pH+ 
                      sample_data(f1)$C.N +
                      sample_data(f1)$Parent_material+
                      sample_data(f1)$LDMC,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Late vs Forest"
q1$Diversity = "P-cycling"
q1$Factor = factors2
q1$Covariates = 1
k6 = q1


r6 = rbind(k1,k2,k3,k4,k5,k6)

```



```{r effect size bacteria ncycling}

#WITHOUT COVARIATES
f1 = ps_ncyc_bac %>% 
  ps_filter(Succession %in% c("Managed","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")

s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
s1
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Managed vs Forest"
q1$Diversity = "P-cycling"
q1$Factor = factors
q1$Covariates = 0
k1 = q1

#Recent
f1 = ps_ncyc_bac %>% 
  ps_filter(Succession %in% c("Recent","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Recent vs Forest"
q1$Diversity = "P-cycling"
q1$Factor = factors
q1$Covariates = 0
k2 = q1

#Late
f1 = ps_ncyc_bac %>% 
  ps_filter(Succession %in% c("Late","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Late vs Forest"
q1$Diversity = "P-cycling"
q1$Factor = factors
q1$Covariates = 0
k3 = q1


#WITH COVARIATES
#Managed
f1 = ps_ncyc_bac %>% 
  ps_filter(Succession %in% c("Managed","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")

s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession +
                      sample_data(f1)$pH+ 
                      sample_data(f1)$C.N +
                      sample_data(f1)$Parent_material+
                      sample_data(f1)$LDMC,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
s1
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Managed vs Forest"
q1$Diversity = "P-cycling"
q1$Factor = factors2
q1$Covariates = 1
k4 = q1

#Recent
f1 = ps_ncyc_bac %>% 
  ps_filter(Succession %in% c("Recent","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession +
                      sample_data(f1)$pH+ 
                      sample_data(f1)$C.N +
                      sample_data(f1)$Parent_material+
                      sample_data(f1)$LDMC,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Recent vs Forest"
q1$Diversity = "P-cycling"
q1$Factor = factors2
q1$Covariates = 1
k5 = q1

#Late
f1 = ps_ncyc_bac %>% 
  ps_filter(Succession %in% c("Late","Forest")) %>%
  ps_filter(n()>1, .by = Par) %>%
  clean_pq()

dm = f1 %>% phyloseq::distance(method = "bray")
s1 = vegan::adonis2(dm ~ sample_data(f1)$Succession +
                      sample_data(f1)$pH+ 
                      sample_data(f1)$C.N +
                      sample_data(f1)$Parent_material+
                      sample_data(f1)$LDMC,
                      nperm=9999,
                    by = "margin",
                    na.action = na.omit)
q1 = adonis_OmegaSq(s1, partial = T)
q1$Community = "Bacteria"
q1$R2 = s1$R2
q1$Comparison = "Late vs Forest"
q1$Diversity = "P-cycling"
q1$Factor = factors2
q1$Covariates = 1
k6 = q1


r7 = rbind(k1,k2,k3,k4,k5,k6)

```


```{r Merge and write results}

res_es = rbind(r1,r2,r3,r4,r5,r6,r7)

write.table(res_es, file = "./results/es_bd.txt")

```


                      CHECK FOR DIFFERENCES IN STAND AGE AND EVENNESS
###########################################################################################

```{r Fungi }

#Taxonomic diversity
ps_fungi %>%
  ps_filter(Succession == "Forest") %>%
  adonis_pq("Age", na_remove = T)

ps_fungi %>%
  ps_filter(Succession == "Forest") %>%
  adonis_pq("Evenness", na_remove = T)

ps_fungi %>%
  ps_filter(Succession == "Forest") %>%
  adonis_pq("Evenness*Age", na_remove = T)

#C-cycling diversity
ps_caz_fun %>%
  ps_filter(Succession == "Forest") %>%
  adonis_pq("Age", na_remove = T)

ps_caz_fun %>%
  ps_filter(Succession == "Forest") %>%
  adonis_pq("Evenness", na_remove = T)

ps_caz_fun %>%
  ps_filter(Succession == "Forest") %>%
  adonis_pq("Evenness*Age", na_remove = T)

#P-cycling diversity
ps_pcyc_fun %>%
  ps_filter(Succession == "Forest") %>%
  adonis_pq("Age", na_remove = T)

ps_pcyc_fun %>%
  ps_filter(Succession == "Forest") %>%
  adonis_pq("Evenness", na_remove = T)

ps_pcyc_fun %>%
  ps_filter(Succession == "Forest") %>%
  adonis_pq("Evenness*Age", na_remove = T)

```

```{r Bacteria }

#Taxonomic diversity
ps_bacteria %>%
  ps_filter(Succession == "Forest") %>%
  adonis_pq("Age", na_remove = T)

ps_bacteria %>%
  ps_filter(Succession == "Forest") %>%
  adonis_pq("Evenness", na_remove = T)

ps_bacteria %>%
  ps_filter(Succession == "Forest") %>%
  adonis_pq("Evenness*Age", na_remove = T)

#C-cycling diversity
ps_caz_bac %>%
  ps_filter(Succession == "Forest") %>%
  adonis_pq("Age", na_remove = T)

ps_caz_bac %>%
  ps_filter(Succession == "Forest") %>%
  adonis_pq("Evenness", na_remove = T)

ps_caz_bac %>%
  ps_filter(Succession == "Forest") %>%
  adonis_pq("Evenness*Age", na_remove = T)

#P-cycling diversity
ps_pcyc_bac %>%
  ps_filter(Succession == "Forest") %>%
  adonis_pq("Age", na_remove = T)

ps_pcyc_bac %>%
  ps_filter(Succession == "Forest") %>%
  adonis_pq("Evenness", na_remove = T)

ps_pcyc_bac %>%
  ps_filter(Succession == "Forest") %>%
  adonis_pq("Evenness*Age", na_remove = T)


#N-cycling diversity
ps_ncyc_bac %>%
  ps_filter(Succession == "Forest") %>%
  adonis_pq("Age", na_remove = T)

ps_ncyc_bac %>%
  ps_filter(Succession == "Forest") %>%
  adonis_pq("Evenness", na_remove = T)

ps_ncyc_bac %>%
  ps_filter(Succession == "Forest") %>%
  adonis_pq("Evenness*Age", na_remove = T)

```




