---
title: "niche_overlap_cazyme_ver1"
author: "Tord Ranheim"
date: "2024-10-29"
output: html_document
---


Split by landuse


Source:
https://github.com/DamienFinn/MicroNiche_Vignette/blob/master/MicroNiche_vignette.pdf


```{r setup, include=FALSE}
knitr::opts_chunk$set(dev = "png",
                      dpi=300,
                      echo=F,
                      cache=T, echo=T)
```


```{r}
rm(list=ls())
```


Libraries
```{r}
library(stringr)
library(dplyr)
library(tidyverse)
library(phyloseq)
library(metagMisc)
library(MicroNiche)
```


Data loading
```{r Load phyloseq objects}

#fungal cazymes
ps_fungi = readRDS("/Users/tranheim/Desktop/Studier/Remiil-NFI/Study 1/Processing/Microbial/Metagenome/Processing/CAZymes/Results/ps_caz_fun_rar.RDS")

#bacterial cazyme
ps_bacteria = readRDS("/Users/tranheim/Desktop/Studier/Remiil-NFI/Study 1/Processing/Microbial/Metagenome/Processing/CAZymes/Results/ps_caz_bac_rar.RDS")

```


```{r Load metadata file}

metadata = read.table("/Users/tranheim/Desktop/Studier/Remiil-NFI/Study 1/Processing/Metadata/Result/metadata.txt")
row.names(metadata) = metadata$Sample

```


```{r load mapping file}

map = read.table("/Users/tranheim/Desktop/Databases/Bacteria/Genes/cazymes/cazyme_substrates.txt", header = T)

unique(map$Substrate)

#Fix groups
map = map %>% mutate(Substrate = case_when(
Substrate == "O.Plant.Polysac." ~ "Other_Polysaccharides",
Substrate == "O.Ani.Polysac." ~ "Other_Polysaccharides",
TRUE ~ as.character(Substrate)))


```


                                      PREPARE DATAFRAMES
###################################################################################################

```{r Split phyloseq}

f1 = ps_fungi %>% phyloseq_sep_variable("Succession")
f2 = ps_bacteria %>% phyloseq_sep_variable("Succession")

```


                                CALCULATE LEVIN'S NICHE OVERLAP FUNGI
###################################################################################################

```{r niche overlap managed fungi}

#Calculate niche overlap
z1 = as.data.frame(as.matrix(f1[["Managed"]]@otu_table))
z1$CAZy = row.names(z1)
z1 = z1[,c(dim(z1)[2],1:dim(z1)[2]-1)]
z2 = levins.overlap(z1)

#Subset to significant genes only
names = z2$LO1.2
rows <- grep("\\*", z2[,1], invert = TRUE)
z2 = z2[rows,]
z2 = z2[z2$LO1.2]
names = z2$LO1.2
row.names(z2) = names(z2)
lno_cazyme_fungi_managed = z2

#Write results
write.table(lno_cazyme_fungi_managed, file = "lno_cazyme_fungi_managed.txt")


#Split by substrate groups
z1 = lno_cazyme_fungi_managed
names = row.names(z1)
v1 = melt(z1) %>% filter(value < 1)
v1$substrate <- map$Substrate[match(v1$variable, map$CAZy)]

#Replace doubles
v1 = v1 %>%
  mutate(substrate = case_when(
    str_sub(variable, 1, 4) == "GH5_" ~ "Cellulose",
        str_sub(variable, 1, 5) == "G13_" ~ "Starch_Glycogen",
       str_sub(variable, 1, 5) == "G30_" ~ "Xylan",
    str_sub(variable, 1, 5) == "GH43_" ~ "Other_Polysaccharides",
    str_sub(variable, 1, 4) == "AA1_" ~ "Lignin",
    str_sub(variable, 1, 4) == "AA2_" ~ "Lignin",
    str_sub(variable, 1, 4) == "AA3_" ~ "Lignin",
    str_sub(variable, 1, 4) == "PL1_" ~ "Pectin",
    str_sub(variable, 1, 4) == "PL3_" ~ "Pectin",
    str_sub(variable, 1, 4) == "PL9_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL10_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL11_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL22_" ~ "Pectin",
    TRUE ~ substrate
  ))

#Replace NAs
v1$substrate[is.na(v1$substrate)] <- "Not listed"
v1$Succession = "Managed"
v1$Sucession_numeric = 1
v1$Community = "Fungi"

s1 = v1

```


```{r niche overlap recent fungi}

#Calculate niche overlap
z1 = as.data.frame(as.matrix(f1[["Recent"]]@otu_table))
z1$CAZy = row.names(z1)
z1 = z1[,c(dim(z1)[2],1:dim(z1)[2]-1)]
z2 = levins.overlap(z1)

#Subset to significant genes only
names = z2$LO1.2
rows <- grep("\\*", z2[,1], invert = TRUE)
z2 = z2[rows,]
z2 = z2[z2$LO1.2]
names = z2$LO1.2
row.names(z2) = names(z2)
lno_cazyme_fungi_recent = z2

#Write results
write.table(lno_cazyme_fungi_recent, file = "lno_cazyme_fungi_recent.txt")


#Split by substrate groups
z1 = lno_cazyme_fungi_recent
names = row.names(z1)
v1 = melt(z1) %>% filter(value < 1)
v1$substrate <- map$Substrate[match(v1$variable, map$CAZy)]

#Replace doubles
v1 = v1 %>%
  mutate(substrate = case_when(
    str_sub(variable, 1, 4) == "GH5_" ~ "Cellulose",
        str_sub(variable, 1, 5) == "G13_" ~ "Starch_Glycogen",
       str_sub(variable, 1, 5) == "G30_" ~ "Xylan",
    str_sub(variable, 1, 5) == "GH43_" ~ "Other_Polysaccharides",
    str_sub(variable, 1, 4) == "AA1_" ~ "Lignin",
    str_sub(variable, 1, 4) == "AA2_" ~ "Lignin",
    str_sub(variable, 1, 4) == "AA3_" ~ "Lignin",
    str_sub(variable, 1, 4) == "PL1_" ~ "Pectin",
    str_sub(variable, 1, 4) == "PL3_" ~ "Pectin",
    str_sub(variable, 1, 4) == "PL9_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL10_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL11_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL22_" ~ "Pectin",
    TRUE ~ substrate
  ))

#Replace NAs
v1$substrate[is.na(v1$substrate)] <- "Not listed"
v1$Succession = "Recent"
v1$Sucession_numeric = 2
v1$Community = "Fungi"

s2 = v1

```


```{r niche overlap late fungi}

#Calculate niche overlap
z1 = as.data.frame(as.matrix(f1[["Late-term"]]@otu_table))
z1$CAZy = row.names(z1)
z1 = z1[,c(dim(z1)[2],1:dim(z1)[2]-1)]
z2 = levins.overlap(z1)

#Subset to significant genes only
names = z2$LO1.2
rows <- grep("\\*", z2[,1], invert = TRUE)
z2 = z2[rows,]
z2 = z2[z2$LO1.2]
names = z2$LO1.2
row.names(z2) = names(z2)
lno_cazyme_fungi_late = z2

#Write results
write.table(lno_cazyme_fungi_late, file = "lno_cazyme_fungi_late.txt")


#Split by substrate groups
z1 = lno_cazyme_fungi_late
names = row.names(z1)
v1 = melt(z1) %>% filter(value < 1)
v1$substrate <- map$Substrate[match(v1$variable, map$CAZy)]

#Replace doubles
v1 = v1 %>%
  mutate(substrate = case_when(
    str_sub(variable, 1, 4) == "GH5_" ~ "Cellulose",
        str_sub(variable, 1, 5) == "G13_" ~ "Starch_Glycogen",
       str_sub(variable, 1, 5) == "G30_" ~ "Xylan",
    str_sub(variable, 1, 5) == "GH43_" ~ "Other_Polysaccharides",
    str_sub(variable, 1, 4) == "AA1_" ~ "Lignin",
    str_sub(variable, 1, 4) == "AA2_" ~ "Lignin",
    str_sub(variable, 1, 4) == "AA3_" ~ "Lignin",
    str_sub(variable, 1, 4) == "PL1_" ~ "Pectin",
    str_sub(variable, 1, 4) == "PL3_" ~ "Pectin",
    str_sub(variable, 1, 4) == "PL9_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL10_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL11_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL22_" ~ "Pectin",
    TRUE ~ substrate
  ))

#Replace NAs
v1$substrate[is.na(v1$substrate)] <- "Not listed"
v1$Succession = "Late"
v1$Sucession_numeric = 3
v1$Community = "Fungi"

s3 = v1

```

```{r niche overlap forest fungi}

#Calculate niche overlap
z1 = as.data.frame(as.matrix(f1[["Forest"]]@otu_table))
z1$CAZy = row.names(z1)
z1 = z1[,c(dim(z1)[2],1:dim(z1)[2]-1)]
z2 = levins.overlap(z1)

#Subset to significant genes only
names = z2$LO1.2
rows <- grep("\\*", z2[,1], invert = TRUE)
z2 = z2[rows,]
z2 = z2[z2$LO1.2]
names = z2$LO1.2
row.names(z2) = names(z2)
lno_cazyme_fungi_forest = z2

#Write results
write.table(lno_cazyme_fungi_forest, file = "lno_cazyme_fungi_forest.txt")


#Split by substrate groups
z1 = lno_cazyme_fungi_forest
names = row.names(z1)
v1 = melt(z1) %>% filter(value < 1)
v1$substrate <- map$Substrate[match(v1$variable, map$CAZy)]

#Replace doubles
v1 = v1 %>%
  mutate(substrate = case_when(
    str_sub(variable, 1, 4) == "GH5_" ~ "Cellulose",
        str_sub(variable, 1, 5) == "G13_" ~ "Starch_Glycogen",
       str_sub(variable, 1, 5) == "G30_" ~ "Xylan",
    str_sub(variable, 1, 5) == "GH43_" ~ "Other_Polysaccharides",
    str_sub(variable, 1, 4) == "AA1_" ~ "Lignin",
    str_sub(variable, 1, 4) == "AA2_" ~ "Lignin",
    str_sub(variable, 1, 4) == "AA3_" ~ "Lignin",
    str_sub(variable, 1, 4) == "PL1_" ~ "Pectin",
    str_sub(variable, 1, 4) == "PL3_" ~ "Pectin",
    str_sub(variable, 1, 4) == "PL9_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL10_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL11_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL22_" ~ "Pectin",
    TRUE ~ substrate
  ))

#Replace NAs
v1$substrate[is.na(v1$substrate)] <- "Not listed"
v1$Succession = "Forest"
v1$Sucession_numeric = 4
v1$Community = "Fungi"

s4 = v1

```

```{r Merge fungi dataframes}

res_fungi = rbind(s1,s2,s3,s4)

```




                              CALCULATE LEVIN'S NICHE OVERLAP BACTERIA
###################################################################################################

```{r niche overlap managed}

#Calculate niche overlap
z1 = as.data.frame(as.matrix(f2[["Managed"]]@otu_table))
z1$CAZy = row.names(z1)
z1 = z1[,c(dim(z1)[2],1:dim(z1)[2]-1)]
z2 = levins.overlap(z1)

#Subset to significant genes only
names = z2$LO1.2
rows <- grep("\\*", z2[,1], invert = TRUE)
z2 = z2[rows,]
z2 = z2[z2$LO1.2]
names = z2$LO1.2
row.names(z2) = names(z2)
lno_cazyme_bacteria_managed = z2

#Write results
write.table(lno_cazyme_bacteria_managed, file = "lno_cazyme_bacteria_managed.txt")


#Split by substrate groups
z1 = lno_cazyme_bacteria_managed
names = row.names(z1)
v1 = melt(z1) %>% filter(value < 1)
v1$substrate <- map$Substrate[match(v1$variable, map$CAZy)]

#Replace doubles
v1 = v1 %>%
  mutate(substrate = case_when(
    str_sub(variable, 1, 4) == "GH5_" ~ "Cellulose",
        str_sub(variable, 1, 5) == "G13_" ~ "Starch_Glycogen",
       str_sub(variable, 1, 5) == "G30_" ~ "Xylan",
    str_sub(variable, 1, 5) == "GH43_" ~ "Other_Polysaccharides",
    str_sub(variable, 1, 4) == "AA1_" ~ "Lignin",
    str_sub(variable, 1, 4) == "AA2_" ~ "Lignin",
    str_sub(variable, 1, 4) == "AA3_" ~ "Lignin",
    str_sub(variable, 1, 4) == "PL1_" ~ "Pectin",
    str_sub(variable, 1, 4) == "PL3_" ~ "Pectin",
    str_sub(variable, 1, 4) == "PL9_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL10_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL11_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL22_" ~ "Pectin",
    TRUE ~ substrate
  ))

#Replace NAs
v1$substrate[is.na(v1$substrate)] <- "Not listed"
v1$Succession = "Managed"
v1$Sucession_numeric = 1
v1$Community = "Bacteria"

s1 = v1

```


```{r niche overlap recent}

#Calculate niche overlap
z1 = as.data.frame(as.matrix(f2[["Recent"]]@otu_table))
z1$CAZy = row.names(z1)
z1 = z1[,c(dim(z1)[2],1:dim(z1)[2]-1)]
z2 = levins.overlap(z1)

#Subset to significant genes only
names = z2$LO1.2
rows <- grep("\\*", z2[,1], invert = TRUE)
z2 = z2[rows,]
z2 = z2[z2$LO1.2]
names = z2$LO1.2
row.names(z2) = names(z2)
lno_cazyme_bacteria_recent = z2

#Write results
write.table(lno_cazyme_bacteria_recent, file = "lno_cazyme_bacteria_recent.txt")


#Split by substrate groups
z1 = lno_cazyme_bacteria_recent
names = row.names(z1)
v1 = melt(z1) %>% filter(value < 1)
v1$substrate <- map$Substrate[match(v1$variable, map$CAZy)]

#Replace doubles
v1 = v1 %>%
  mutate(substrate = case_when(
    str_sub(variable, 1, 4) == "GH5_" ~ "Cellulose",
        str_sub(variable, 1, 5) == "G13_" ~ "Starch_Glycogen",
       str_sub(variable, 1, 5) == "G30_" ~ "Xylan",
    str_sub(variable, 1, 5) == "GH43_" ~ "Other_Polysaccharides",
    str_sub(variable, 1, 4) == "AA1_" ~ "Lignin",
    str_sub(variable, 1, 4) == "AA2_" ~ "Lignin",
    str_sub(variable, 1, 4) == "AA3_" ~ "Lignin",
    str_sub(variable, 1, 4) == "PL1_" ~ "Pectin",
    str_sub(variable, 1, 4) == "PL3_" ~ "Pectin",
    str_sub(variable, 1, 4) == "PL9_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL10_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL11_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL22_" ~ "Pectin",
    TRUE ~ substrate
  ))

#Replace NAs
v1$substrate[is.na(v1$substrate)] <- "Not listed"
v1$Succession = "Recent"
v1$Sucession_numeric = 2
v1$Community = "Bacteria"

s2 = v1
```


```{r niche overlap late}

#Calculate niche overlap
z1 = as.data.frame(as.matrix(f2[["Late-term"]]@otu_table))
z1$CAZy = row.names(z1)
z1 = z1[,c(dim(z1)[2],1:dim(z1)[2]-1)]
z2 = levins.overlap(z1)

#Subset to significant genes only
names = z2$LO1.2
rows <- grep("\\*", z2[,1], invert = TRUE)
z2 = z2[rows,]
z2 = z2[z2$LO1.2]
names = z2$LO1.2
row.names(z2) = names(z2)
lno_cazyme_bacteria_late = z2

#Write results
write.table(lno_cazyme_bacteria_late, file = "lno_cazyme_bacteria_late.txt")


#Split by substrate groups
z1 = lno_cazyme_bacteria_late
names = row.names(z1)
v1 = melt(z1) %>% filter(value < 1)
v1$substrate <- map$Substrate[match(v1$variable, map$CAZy)]

#Replace doubles
v1 = v1 %>%
  mutate(substrate = case_when(
    str_sub(variable, 1, 4) == "GH5_" ~ "Cellulose",
        str_sub(variable, 1, 5) == "G13_" ~ "Starch_Glycogen",
       str_sub(variable, 1, 5) == "G30_" ~ "Xylan",
    str_sub(variable, 1, 5) == "GH43_" ~ "Other_Polysaccharides",
    str_sub(variable, 1, 4) == "AA1_" ~ "Lignin",
    str_sub(variable, 1, 4) == "AA2_" ~ "Lignin",
    str_sub(variable, 1, 4) == "AA3_" ~ "Lignin",
    str_sub(variable, 1, 4) == "PL1_" ~ "Pectin",
    str_sub(variable, 1, 4) == "PL3_" ~ "Pectin",
    str_sub(variable, 1, 4) == "PL9_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL10_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL11_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL22_" ~ "Pectin",
    TRUE ~ substrate
  ))

#Replace NAs
v1$substrate[is.na(v1$substrate)] <- "Not listed"
v1$Succession = "Late"
v1$Sucession_numeric = 3
v1$Community = "Bacteria"

s3 = v1

```



```{r niche overlap forest bacteria}

#Calculate niche overlap
z1 = as.data.frame(as.matrix(f2[["Forest"]]@otu_table))
z1$CAZy = row.names(z1)
z1 = z1[,c(dim(z1)[2],1:dim(z1)[2]-1)]
z2 = levins.overlap(z1)

#Subset to significant genes only
names = z2$LO1.2
rows <- grep("\\*", z2[,1], invert = TRUE)
z2 = z2[rows,]
z2 = z2[z2$LO1.2]
names = z2$LO1.2
row.names(z2) = names(z2)
lno_cazyme_bacteria_forest = z2

#Write results
write.table(lno_cazyme_bacteria_forest, file = "lno_cazyme_bacteria_forest.txt")


#Split by substrate groups
z1 = lno_cazyme_bacteria_forest
names = row.names(z1)
v1 = melt(z1) %>% filter(value < 1)
v1$substrate <- map$Substrate[match(v1$variable, map$CAZy)]

#Replace doubles
v1 = v1 %>%
  mutate(substrate = case_when(
    str_sub(variable, 1, 4) == "GH5_" ~ "Cellulose",
        str_sub(variable, 1, 5) == "G13_" ~ "Starch_Glycogen",
       str_sub(variable, 1, 5) == "G30_" ~ "Xylan",
    str_sub(variable, 1, 5) == "GH43_" ~ "Other_Polysaccharides",
    str_sub(variable, 1, 4) == "AA1_" ~ "Lignin",
    str_sub(variable, 1, 4) == "AA2_" ~ "Lignin",
    str_sub(variable, 1, 4) == "AA3_" ~ "Lignin",
    str_sub(variable, 1, 4) == "PL1_" ~ "Pectin",
    str_sub(variable, 1, 4) == "PL3_" ~ "Pectin",
    str_sub(variable, 1, 4) == "PL9_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL10_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL11_" ~ "Pectin",
    str_sub(variable, 1, 5) == "PL22_" ~ "Pectin",
    TRUE ~ substrate
  ))

#Replace NAs
v1$substrate[is.na(v1$substrate)] <- "Not listed"
v1$Succession = "Forest"
v1$Sucession_numeric = 4
v1$Community = "Bacteria"

s4 = v1

```



```{r Merge bacteria dataframes}

res_bacteria = rbind(s1,s2,s3,s4)

```


```{r merge and write file}

res_lno_cazyme = rbind(res_fungi,res_bacteria)
write.table(res_lno_cazyme, file = "res_lno_cazyme.txt")

```


